beanie/odm/utils/init.py

```diff
@@ -233,8 +233,9 @@ async def init_document_collection(self, cls):
         # register in the Union Doc
 
         if document_settings.union_doc is not None:
+            name = cls.get_settings().name or cls.__name__
```

Comment:
if `class.__name__` returns original class name instead of None when get_settings().name is not set, we wouldn't need this line.